### PR TITLE
Ammo magazine relabeling

### PR DESCRIPTION
**Pull request by SirRichardFrancis**
## About The Pull Request

Changed the way sprite updates work for ammo magazines and couple of guns. Ability to recolor magazines with hand labeler.
Added missing sprites for some magazines and corrected sprite names. A lot of that.

<details>
 <summary>
Screenshots
 </summary>

image

image

</details>

## Why It's Good For The Game

Dunno, looks cool.

## Changelog
:cl:
add: ammo magazine recoloring with hand labeler
/:cl:

### CODE REVIEW

**Comment by Voldirs** [comment]
I think changing stripe should be done manually and not automatically so you can still trick someone by selling lethal mags loaded with practice ammo

**Comment by CeUvi** [comment]
finally, bluespace mags

**Comment by SaltyOldBull** [comment]
> I think changing stripe should be done manually and not automatically so you can still trick someone by selling lethal mags loaded with practice ammo

agreed, changing the ammo inside the magazine is a very viable tactic and should be kept as is. Being able to change it manually would be an issue, however.

**Comment by SirRichardFrancis** [comment]
> I think changing stripe should be done manually and not automatically so you can still trick someone by selling lethal mags loaded with practice ammo

Insert one lethal casing first and fill the rest with practice ammo.
Don't even need a bunch of lethal mags, even an empty casing will be enough.
Furthermore, with manual sprite changing that kind of shenanigans will hardly be possible at all.

> finally, bluespace mags

How is that bluespace? Magazine capacity is untouched.
And not like people forming walls/rods/whatever out of metal sheets with their bare hands is an issue, so why this would be?

> agreed, changing the ammo inside the magazine is a very viable tactic and should be kept as is. Being able to change it manually would be an issue, however.

A very viable tactic? Eh? Would be an issue how exactly?

**Comment by Voldirs** [comment]
>Insert one lethal casing first and fill the rest with practice ammo.

Come on, that's just bad, mags changing color stripes automatically could be just a simplification, but why? We've got pill bottles that need to be relabeled manually, We've got chem bottles that don't change their names automatically, why mags should?

**Comment by SirRichardFrancis** [comment]
> Come on, that's just bad, mags changing color stripes automatically could be just a simplification, but why? We've got pill bottles that need to be relabeled manually, We've got chem bottles that don't change their names automatically, why mags should?

Come on, that's just bad, more features and stuff would be just awesome, but why? We've got bay leftovers that need to be removed manually, We've got kotmap that won't be implemented any time soon, why anything else should?

**Comment by Kegdo** [comment]
> Come on, that's just bad, more features and stuff would be just awesome, but why? We've got bay leftovers that need to be removed manually, We've got kotmap that won't be implemented any time soon, why anything else should?

actually seething, I don't like this one because putting blank rounds into lethal-labeled mags was fun and peak clowning

**Comment by CeUvi** [comment]
kirov is slowly turning into seethe pull cope request lmao

**Comment by Humonitarian** [comment]
I do not like the new features this pr brings

**Comment by SirRichardFrancis** [comment]
Half the complaints have no arguments at all, and those that have is questionable at best imo.
But that's not a big deal, automatic change of magazine name and color just looks like a nicer option to me.
Would changing all that via verb satisfy you all?

**Comment by Humonitarian** [comment]
Just doesn't feel all that immersive to me, how we can change their colour at will.

**Comment by CeUvi** [comment]
maybe have some kind of machine (maybe even portable) to color mags

**Comment by Kegdo** [comment]
> maybe make some kind of machine (maybe even portable) to color mags

literally fucking crayons

**Comment by CeUvi** [comment]
> literally fucking crayons

that works

**Comment by Humonitarian** [comment]
> > maybe make some kind of machine (maybe even portable) to color mags
> 
> literally fucking crayons

Unironically make it crayons, I'm serious

Especially if you can make it in every crayon colour

**Comment by SirRichardFrancis** [comment]
> Especially if you can make it in every crayon colour

Well... yes. But we have certain pattern for mags colors/names and there is less(for regular mags)/more(for drum mags) patterns than crayon colors.
So if mag color is defined by crayon color - those patterns thrown away for no good reason. But if any crayon can paint in any color, that's also odd.
By pattern I mean green standing for high velocity/slugs, blue for rubber/beanbags, orange for rubber/incendiary etc.

**Comment by ValoTheValo** [comment]
> Just doesn't feel all that immersive to me, how we can change their colour at will.

Have you ever heard of coloured tape, cloth or CRAYONS

**Comment by Kegdo** [comment]
image




ok then do it

**Comment by Humonitarian** [comment]
> image
> 
> ok then do it

this

**Comment by Humonitarian** [comment]
> > Just doesn't feel all that immersive to me, how we can change their colour at will.
> 
> Have you ever heard of coloured tape, cloth or CRAYONS

We can change the colour of crayons? Since when?

**Comment by greenteaguzzler** [comment]
Sad to see automatic go :(

**Comment by SirRichardFrancis** [comment]
> Sad to see automatic go :(

image

File: code/modules/projectiles/guns/projectile/automatic/maxim.dm [hunk 1]
```diff
@@ -38,18 +38,14 @@
...
 	..()
-
-	var/iconstring = initial(icon_state)
 	var/itemstring = ""
+	cut_overlays()
 
-	if (ammo_magazine)
-		iconstring += "[ammo_magazine? "-full[ammo_magazine.ammo_color]": ""]"
+	if(ammo_magazine)
+		add_overlays("mag[ammo_magazine.ammo_label_string]")
```
Comment (SirRichardFrancis) [inline code/modules/projectiles/guns/projectile/automatic/maxim.dm:45]:
```suggestion
		iconstring += "mag[ammo_magazine.ammo_label_string]"
```

File: code/modules/projectiles/guns/projectile/automatic/c20r.dm [hunk 2]
```diff
@@ -38,9 +38,12 @@
 /obj/item/gun/projectile/automatic/c20r/update_icon()
 	cut_overlays()
 	icon_state = "[initial(icon_state)][silenced ? "_s" : ""]"
+
 	if(ammo_magazine)
-		overlays += "mag[silenced ? "_s" : ""][ammo_magazine.ammo_color]"
-	if (!ammo_magazine || !length(ammo_magazine.stored_ammo))
+		add_overlays("mag[silenced ? "_s" : ""][ammo_magazine.ammo_label_string]")
```
Comment (SirRichardFrancis) [inline code/modules/projectiles/guns/projectile/automatic/c20r.dm:43]:
```suggestion
		overlays += "mag[silenced ? "_s" : ""][ammo_magazine.ammo_label_string]"
```

File: code/modules/projectiles/guns/projectile/automatic/maxim.dm [hunk 3]
```diff
@@ -38,18 +38,14 @@
...
 	..()
-
-	var/iconstring = initial(icon_state)
 	var/itemstring = ""
+	cut_overlays()
 
-	if (ammo_magazine)
-		iconstring += "[ammo_magazine? "-full[ammo_magazine.ammo_color]": ""]"
+	if(ammo_magazine)
+		iconstring += "mag[ammo_magazine.ammo_label_string]"
```
Comment (SirRichardFrancis) [inline code/modules/projectiles/guns/projectile/automatic/maxim.dm:45]:
```suggestion
		overlays += "mag[ammo_magazine.ammo_label_string]"
```

**Review by Firefox13 — APPROVED** [review]

**Pull request merged**